tests/tracer/test_trace_utils.py

```diff
@@ -109,3 +109,21 @@ def test_set_http_meta(span, config, method, url, status_code):
             assert span.error == 0
     else:
         assert http.STATUS_CODE not in span.meta
+
+
+@pytest.mark.parametrize(
+    "error_codes,status_code,error",
+    [
+        ("404-400", 400, 1),
+        ("400-404", 400, 1),
+        ("400-404", 500, 0),
+        ("500-520", 530, 0),
+        ("500-550         ", 530, 1),
+        ("400-404,419", 419, 1),
+        ("400,401,403", 401, 1),
```

Comment:
Good to add a test here for an error code string that includes non-ints, e.g. `"400-404,X"`.